lib/Compiler.cpp

```diff
@@ -1378,14 +1378,18 @@ ClspvError clspvCompileFromSourcesString(
   err =
       clspv::CompileFromSourcesString(vPrograms, sOptions, &binary, &buildLog);
 
-  if (!buildLog.empty()) {
-    // Alloc and copy backing mem for build log
-    *output_log = static_cast<char *>(std::malloc(buildLog.size() + 1));
-    if (*output_log == NULL) {
-      return CLSPV_OUT_OF_HOST_MEM;
+  if (output_log) {
```

Comment:
```suggestion
  if (output_log != NULL) {
```
`output_log` is a pointer, not a boolean, let's keep thing as readable as possible by avoiding boolean operators on pointers

Reply:
Changed to `output_log != NULL`